File: Trabalhos-Práticos/TP1/Questão 1 - Palindromo - Java/Palindromo.java
public class Palindromo {
    public static void main(String[] args) {
        MyIO.setCharset("UTF-8");
        String string = MyIO.readLine();

        while (!(ehFIM(string))) {
            boolean ehPalindromo = true;
            int contador = string.length();

            for (int i = 0; i < contador / 2; i++) {
                if (string.charAt(i) != string.charAt(contador - i - 1)) {
                    ehPalindromo = false;
                   
                }
            }

            if(ehPalindromo) {
                MyIO.println("SIM");
            } else {
                MyIO.println("NAO");
            }

            string = MyIO.readLine();
        }

       
    }

     public static boolean ehFIM(String string) {
        boolean isFim = false;
        
        if (string.length() == 3) {
            if (string.charAt(0) == 'F' && string.charAt(1) == 'I' && string.charAt(2) == 'M') {
                isFim = true;
            }
        }
        
        return isFim;
    }
}

File: Trabalhos-Práticos/TP1/Questão 02 - Palindromo - C/palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <locale.h>



#define TAM 100

bool ehPalindromo(char string[]) {
    bool palindromo = true;
    int tamanho = strlen(string);

    for(int i=0;i<tamanho/2;i++) {
        if(string[i] != string[tamanho - 1 -i]) {
            palindromo = false;
        }
    
    }
    return palindromo;
}

int main() {
    setlocale(LC_ALL, "pt_BR.UTF-8");

    char entrada[TAM];

    scanf(" ^[%\n\r]",entrada);
    getchar();

    while(strcmp(entrada, "FIM")!= 0) {
        if(ehPalindromo(entrada)) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }

        scanf(" ^[%\n\r]",entrada);
        getchar();

        
    }

    return 0;   

}